crates/db_views/src/post_view.rs

```diff
@@ -409,6 +419,11 @@ impl<'a> PostQueryBuilder<'a> {
       query = query.filter(post_read::id.is_null());
     }
 
+    // Filter out the rows with missing languages
+    if self.my_local_user_id.is_some() {
+      query = query.filter(local_user_language::id.is_not_null());
+    }
```

Comment:
I believe the if body needs to be something like this:
```

      query = query.filter(sql(
        "TRUE GROUP BY (post.id, person.id, community.id, community_person_ban.id, post_aggregates.id, \
        community_follower.id, post_saved.id, post_read.id, person_block.id, post_like.score, language.id,\
        local_user_language.id, community_block.person_id) \
        HAVING count(local_user_language.id) = 0 or local_user_language.id is not null"))
```

Main part is the having clause, if all local_user_language.id values are null, dont filter. Unfortunately it doesnt work correctly, even if a user has one language configured, it returns posts for all languages. The group by is necessary because otherwise postgres throws errors like `column "community_block.person_id" must appear in the GROUP BY clause or be used in an aggregate function`. And having/group by are only implemented in diesel 2.0.

Reply:
I'll take a look this weekend and try to get it working.